cmd/cluster/status.go

```diff
@@ -0,0 +1,199 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"unicode"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	k8spkg "github.com/openshift/osdctl/pkg/k8s"
+	awsprovider "github.com/openshift/osdctl/pkg/provider/aws"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+// statusOptions defines the struct for running status command
+type statusOptions struct {
+	k8sclusterresourcefactory k8spkg.ClusterResourceFactoryOptions
+	output                    string
+	verbose                   bool
+
+	genericclioptions.IOStreams
+}
+
+// newCmdStatus implements the status command to describe cluster status
+func newCmdStatus(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	ops := newStatusOptions(streams, flags)
+	statusCmd := &cobra.Command{
+		Use:               "status",
+		Short:             "Describes status of cluster nodes and provides other cluster vitals.",
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	ops.k8sclusterresourcefactory.AttachCobraCliFlags(statusCmd)
+	statusCmd.Flags().StringVarP(&ops.output, "out", "o", "default", "Output format [default | json | env]")
+	statusCmd.Flags().BoolVarP(&ops.verbose, "verbose", "", false, "Verbose output")
+
+	return statusCmd
+}
+
+func newStatusOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *statusOptions {
+	return &statusOptions{
+		k8sclusterresourcefactory: k8spkg.ClusterResourceFactoryOptions{
+			Flags: flags,
+		},
+		IOStreams: streams,
+	}
+}
+
+func (o *statusOptions) complete(cmd *cobra.Command, _ []string) error {
+	var err error
+
+	k8svalid, err := o.k8sclusterresourcefactory.ValidateIdentifiers()
+	if !k8svalid {
+		if err != nil {
+			cmdutil.PrintErrorWithCauses(err, o.ErrOut)
+			return err
+		}
+
+	}
+
+	awsvalid, err := o.k8sclusterresourcefactory.Awscloudfactory.ValidateIdentifiers()
+	if !awsvalid {
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *statusOptions) run() error {
+
+	// This call gets the availability zone of the cluster, as well as the numbers of compute, infra, and master nodes.
+	az, compute, infra, master, err := ocmDescribe(o.k8sclusterresourcefactory.ClusterID)
```

Comment:
We should add an explicit check to see what the cloud provider of an cluster is, in order to explicitly fail for non-AWS clusters so SREs running this on, say, a GCP cluster won't spend time debugging why their AWS access isn't working.

Reply:
The command will throw an "Error: cluster provider is not AWS", the check is on L194. I agree that the error message could be better, for example, "This command is supported only for AWS clusters. The command is not supported for <cluster provider extracted by the command> clusters."

Reply:
I added the change.